lib/elixir/lib/kernel/special_forms.ex

```diff
@@ -1408,6 +1408,19 @@ defmodule Kernel.SpecialForms do
   filters or inside the block, are not reflected outside of the
   comprehension.
 
+  Variable assignment based on `=` is treated as filters, and 
+  if value of the expression is falsey (`nil` and `false`), it would be filtered out.
+  in case variable assignment don't want to be treated as filters, using assignment via generators would be preferred.
```

Comment:
```suggestion
  in case you don't want variable assignment to be treated as filters, use a `<-` generator
  and a single-element list literal on its right-hand side.
```